Providers/LoginRecordExtension.swift

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Storage
+import AuthenticationServices
+
+@available(iOS 12, *)
+extension LoginRecord {
+    open var passwordCredentialIdentity: ASPasswordCredentialIdentity {
+        let serviceIdentifier = ASCredentialServiceIdentifier(identifier: self.hostname, type: .URL)
+        return ASPasswordCredentialIdentity(serviceIdentifier: serviceIdentifier, user: self.username, recordIdentifier: self.id)
+    }
+    
+    open var passwordCredential: ASPasswordCredential {
+        return ASPasswordCredential(user: self.username, password: self.password)
+    }
+}
+
+extension LoginRecord: Comparable {
+    public static func < (lhs: LoginRecord, rhs: LoginRecord) -> Bool {
+        lhs.hostname.titleFromHostname < rhs.hostname.titleFromHostname
+    }
+}
+
+extension LoginRecord: Equatable, Hashable {
+    public static func == (lhs: LoginRecord, rhs: LoginRecord) -> Bool {
+        return lhs.id == rhs.id && lhs.hostname == rhs.hostname && lhs.credentials == rhs.credentials
+    }
+    public func hash(into hasher: inout Hasher) {
+        hasher.combine(self.id)
+    }
+}
```

Comment:
This looks pretty harmless but it shows some warnings now - maybe because of the new App Services library?

```
/Users/stefan/Mozilla/firefox-ios/Providers/LoginRecordExtension.swift:26:24: warning: conformance of 'Login' to protocol 'Equatable' was already stated in the type's module 'MozillaAppServices'
extension LoginRecord: Equatable, Hashable {
                       ^
/Users/stefan/Mozilla/firefox-ios/Providers/LoginRecordExtension.swift:27:24: note: operator function '==' will not be used to satisfy the conformance to 'Equatable'
    public static func == (lhs: LoginRecord, rhs: LoginRecord) -> Bool {
                       ^
MozillaAppServices.Login:1:11: note: 'Login' declares conformance to protocol 'Equatable' here
extension Login : Equatable, Hashable {
          ^
/Users/stefan/Mozilla/firefox-ios/Providers/LoginRecordExtension.swift:26:35: warning: conformance of 'Login' to protocol 'Hashable' was already stated in the type's module 'MozillaAppServices'
extension LoginRecord: Equatable, Hashable {
                                  ^
/Users/stefan/Mozilla/firefox-ios/Providers/LoginRecordExtension.swift:30:17: note: instance method 'hash(into:)' will not be used to satisfy the conformance to 'Hashable'
    public func hash(into hasher: inout Hasher) {
                ^
MozillaAppServices.Login:1:11: note: 'Login' declares conformance to protocol 'Hashable' here
extension Login : Equatable, Hashable {
          ^
```

Reply:
@razvanlitianu thoughts?